tools/objinfo.ml

```diff
@@ -247,19 +247,19 @@ let read_dyn_header filename ic =
   try
     try_finally
       (fun () ->
-        let rc = Sys.command (sprintf "%s %s > %s"
-                                (Filename.quote helper)
-                                (Filename.quote filename)
-                                tempfile) in
-        if rc <> 0 then failwith "cannot read";
-        let tc = Scanf.Scanning.from_file tempfile in
-        try_finally
-          (fun () ->
-            let ofs = Scanf.bscanf tc "%Ld" (fun x -> x) in
-            LargeFile.seek_in ic ofs;
-            Some(input_value ic : dynheader))
-          (fun () -> Scanf.Scanning.close_in tc))
-      (fun () -> remove_file tempfile)
+         let rc = Sys.command (sprintf "%s %s > %s"
+                                 (Filename.quote helper)
+                                 (Filename.quote filename)
+                                 tempfile) in
+         if rc <> 0 then failwith "cannot read";
+         let tc = Scanf.Scanning.from_file tempfile in
+         try_finally
+           (fun () ->
+              let ofs = Scanf.bscanf tc "%Ld" (fun x -> x) in
+              LargeFile.seek_in ic ofs;
+              Some(input_value ic : dynheader))
+           ~always:(fun () -> Scanf.Scanning.close_in tc))
+      ~always:(fun () -> remove_file tempfile)
```

Comment:
The always blocks should go first. (I would also permute the definitions of `helper` and `tempfile` to have `tempfile` opening and closing nearby.)